rtpsender.go

```diff
@@ -86,10 +89,35 @@ func (r *RTPSender) Track() TrackLocal {
 	return r.track
 }
 
-func (r *RTPSender) setTrack(track TrackLocal) {
+// ReplaceTrack replaces the track currently being used as the sender's source with a new TrackLocal.
+// The new track must be of the same media kind (audio, video, etc) and switching the track should not
+// require negotiation.
+func (r *RTPSender) ReplaceTrack(track TrackLocal) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
+
+	if r.hasSent() {
+		if err := r.track.Unbind(TrackLocalContext{
+			id:          r.id,
+			ssrc:        r.ssrc,
+			writeStream: r.rtpWriteStream,
+		}); err != nil {
+			return err
+		}
+	}
+
 	r.track = track
+	if !r.hasSent() || r.track == nil {
```

Comment:
Should we not implement more constraints?, according to spec we should check that track does not difer from current negotiated, also check special cases like audio not being negotiated as 1 channel and replace with 2 channel

Reply:
Yea! Subsequent calls to `Bind` only can use the same codec as the first call https://github.com/pion/webrtc/pull/1527/files#diff-1af62154d0fa7a1013519d5059b0bb1b13f07f25aa4ec4a70bdbc0007e3fd652R100 is a test showing it